vllm/entrypoints/openai/serving_chat.py

```diff
@@ -82,18 +81,17 @@ def __init__(
                 " the parallel_tool_calls client option is preset for "
                 "compatibility reasons, it will be ignored.")
 
-        self.enable_reasoning: bool = enable_reasoning
         self.reasoning_parser: Optional[Callable[[AnyTokenizer],
                                                  ReasoningParser]] = None
-        if self.enable_reasoning:
+        self.enable_reasoning: bool = self.reasoning_parser is not None
```

Comment:
Shall we just remove this completely and instead check if `self.reaoning_parser is not None`?

Reply:
Updated.